Make lint run when directories are passed in

teh-f00l opened the pull request:
Response to issue https://github.com/w3c/web-platform-tests/issues/9879

Allows tester to pass command line arguments to lint specifying directories they want checked

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/9995)
<!-- Reviewable:end -->

teh-f00l committed:
Update lint.py

Response to issue https://github.com/w3c/web-platform-tests/issues/9879

Allows tester to pass command line arguments to lint specifying directories they want checked

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,19 @@ def changed_files(wpt_root):
-        r = os.path.realpath(wpt_root)
-        paths = [os.path.relpath(os.path.realpath(x), r) for x in kwargs["paths"]]
+        #make new list in case there are multiple added directories
+        AllPaths=[]
```
Use `snake_case` for variable names.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,19 @@ def changed_files(wpt_root):
+        AllPaths=[]
+        for path in kwargs.get("paths"):
+            # create extension for called directory and merge it with the existing path
+            extension= path+'/'
```
Use a space before and after operators like `=`.

teh-f00l replied:
kk

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,19 @@ def changed_files(wpt_root):
+        for path in kwargs.get("paths"):
+            # create extension for called directory and merge it with the existing path
+            extension= path+'/'
+            newPath=(str(os.path.realpath(wpt_root))+'/'+ extension)
```
So I feel like this is basically
`abs_path = os.path.realpath(os.path.join(wpt_root, path))`?

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,19 @@ def changed_files(wpt_root):
+            extension= path+'/'
+            newPath=(str(os.path.realpath(wpt_root))+'/'+ extension)
+            # get all files from new path
+            paths = list(all_filesystem_paths(newPath))
```
Do we first want to check if it's a directory?

teh-f00l replied:
ah my bad, I got rid of my if statement. Putting that back in

teh-f00l replied:
Although I am not sure if relpath was sufficient or not

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,19 @@ def changed_files(wpt_root):
+            paths = list(all_filesystem_paths(newPath))
+            # add the extension to the strings for all files from the new path,
+            # so they may be called from the current directory
+            paths=list(map(lambda path : extension + path, paths))
```
A list comprehension is more idomatic here, so this would look like `[extension + path for path in paths]`. That said, it isn't clear what you're doing here. In the clause below the results of `all_filesystem_paths` are used directly, so it seems like it should be fine to do that here too.

teh-f00l replied:
I tried that initially, but it was insufficient because it assumes we are already in the directory passed to ```all_filesystem_paths``` and will not work unless we are already operating from that directory

jgraham reviewed (changes requested):
This looks like a great start, but some changes are required before it can land.

w3c-bots commented:
# Build BROKEN

Started: 2018-03-13 13:27:45
Finished: 2018-03-13 13:42:38


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in py36</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25855)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/352836660)

teh-f00l committed:
Update lint.py

same as before except:
no functional programming
snake_case variables
checking if command line argument is a directory

teh-f00l committed:
Update lint.py

fixed a small variable error, and made the variable names more distinct

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,21 @@ def changed_files(wpt_root):
+        for path in kwargs.get("paths"):
+            if os.path.isdir(path):
+            # create extension for called directory and merge it with the existing path
+                new_Path = os.path.realpath(os.path.join(wpt_root, path))
```
Variable names  should be all lowercase.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,21 @@ def changed_files(wpt_root):
+            if os.path.isdir(path):
+            # create extension for called directory and merge it with the existing path
+                new_Path = os.path.realpath(os.path.join(wpt_root, path))
+                extension = path+'/'
```
So this should use `os.path.sep`, but  I'm not sure why it should be necessary.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,21 @@ def changed_files(wpt_root):
-        r = os.path.realpath(wpt_root)
-        paths = [os.path.relpath(os.path.realpath(x), r) for x in kwargs["paths"]]
+        #make new list in case there are multiple added directories
+        paths=[]
```
Spaces between operators and operands, please.

teh-f00l committed:
Update lint.py

removed caps from variables, put spaces between operators and replaced 'extension' variable with os.path.sep for cross system compatibility

teh-f00l committed:
Update lint.py

updated all_filesystem_paths to work inside the specified directory

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,18 @@ def changed_files(wpt_root):
+        #make new list in case there are multiple added directories
+        paths = []
+        for path in kwargs.get("paths"):
+            if os.path.isdir(path):
```
I think this  `if` is missing an else clause to just add the path to the list of paths.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,18 @@ def changed_files(wpt_root):
+        paths = []
+        for path in kwargs.get("paths"):
+            if os.path.isdir(path):
+            # create extension for called directory and merge it with the existing path
```
This  indent is wrong.

jgraham commented on tools/lint/lint.py:
```diff
@@ -59,9 +59,9 @@ def setup_logging(prefix=False):
 %s: %s"""
 
-def all_filesystem_paths(repo_root):
+def all_filesystem_paths(repo_root, extended_path):
```
"`extended_path`" needs to be optional i.e. the signature should be

`all_filesystem_paths(repo_root, subdir=None):`

Also it should be relative  to `repo_root` so that in the function we do `root = os.path.join(repo_root, subdir) if subdir else repo_root` or something  similar.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,18 @@ def changed_files(wpt_root):
+            # create extension for called directory and merge it with the existing path
+                new_path = os.path.realpath(os.path.join(wpt_root, path))
+                # get all files from new path
+                path_dir = list(all_filesystem_paths(wpt_root, new_Path))
```
Your capitalisation is off here.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,18 @@ def changed_files(wpt_root):
+        for path in kwargs.get("paths"):
+            if os.path.isdir(path):
+            # create extension for called directory and merge it with the existing path
+                new_path = os.path.realpath(os.path.join(wpt_root, path))
```
So this assumes that `path` is relative to the `repo_root` but above it's assumed to be relative to the current working directory. I think the latter is correct.

teh-f00l replied:
The later is correct. It should have been correct here as well, new_path should be the former.

jgraham commented on tools/lint/lint.py:
```diff
@@ -747,8 +747,18 @@ def changed_files(wpt_root):
+                path_dir = list(all_filesystem_paths(wpt_root, new_Path))
+             
+                # add all files from a given directory to our master list of files to check
+                paths = paths + path_dir
```
I think `paths.extend(path_dir)` is more obvious

teh-f00l committed:
Update lint.py

updated all_filesystem_paths to work with or without specified directory arguments

teh-f00l committed:
Update lint.py

added an elif statement to account for files being specified in the command line 
removed excessive comments

teh-f00l committed:
Update lint.py

removed new_paths variable and just put the command line path input into all_filesystem_paths

teh-f00l committed:
Update lint.py

put spaces between '=' all operators

teh-f00l committed:
Update lint.py

made a mistake with my last commit where I added spaces rather than remove them

teh-f00l committed:
Update lint.py

text wrapper didn't work and missed some changes I needed to undo

jgraham committed:
Revert erroneous whitespace change

teh-f00l committed:
Update lint.py

removed the space

teh-f00l committed:
Update lint.py

teh-f00l committed:
Update lint.py

changed the elif isfile check to a simple else statement

teh-f00l committed:
Update lint.py

giving input files, a path relative to the repo_root

teh-f00l committed:
Update test_lint.py

update test_lint to mock files
